scripts/soft-reboot

```diff
@@ -0,0 +1,217 @@
+#!/bin/bash
+DEVPATH="/usr/share/sonic/device"
+REBOOT_CAUSE_FILE="/host/reboot-cause/reboot-cause.txt"
+REBOOT_TIME=$(date)
+REBOOT_METHOD="/sbin/kexec -e"
+LOG_SSD_HEALTH="/usr/local/bin/log_ssd_health"
+
+
+EXIT_SUCCESS=0
+EXIT_FAILURE=1
+EXIT_NOT_SUPPORTED=2
+EXIT_FILE_SYSTEM_FULL=3
+EXIT_NEXT_IMAGE_NOT_EXISTS=4
+
+# Reboot immediately if we run the kdump capture kernel
+VMCORE_FILE=/proc/vmcore
+if [ -e $VMCORE_FILE -a -s $VMCORE_FILE ]; then
+        echo "We have a /proc/vmcore, then we just kdump'ed"
+        echo "User issued 'kdump' command [User: kdump, Time: ${REBOOT_TIME}]" > ${REBOOT_CAUSE_FILE}
+        sync
+        PLATFORM=$(grep -oP 'sonic_platform=\K\S+' /proc/cmdline)
+        if [ ! -z "${PLATFORM}" -a -x ${DEVPATH}/${PLATFORM}/${PLAT_REBOOT} ]; then
+            exec ${DEVPATH}/${PLATFORM}/${PLAT_REBOOT}
+        fi
+        # If no platform-specific reboot tool, just run /sbin/reboot
+        /sbin/reboot
+        echo 1 > /proc/sys/kernel/sysrq
+        echo b > /proc/sysrq-trigger
+fi
+
+REBOOT_USER=$(logname)
+PLATFORM=$(sonic-cfggen -H -v DEVICE_METADATA.localhost.platform)
+ASIC_TYPE=$(sonic-cfggen -y /etc/sonic/sonic_version.yml -v asic_type)
+VERBOSE=no
+EXIT_NEXT_IMAGE_NOT_EXISTS=4
+EXIT_SONIC_INSTALLER_VERIFY_REBOOT=21
+SSD_FW_UPDATE="ssd-fw-upgrade"
+REBOOT_SCRIPT_NAME=$(basename $0)
+REBOOT_TYPE="${REBOOT_SCRIPT_NAME}"
+PLATFORM_PLUGIN="${REBOOT_TYPE}_plugin"
+BOOT_TYPE_ARG="soft"
+TAG_LATEST=yes
+
+function debug()
+{
+    if [[ x"${VERBOSE}" == x"yes" ]]; then
+        echo `date` $@
+    fi
+    logger "$@"
+}
+
+function tag_images()
+{
+    if test -f /usr/local/bin/ctrmgr_tools.py
+    then
+        if [[ x"${TAG_LATEST}" == x"yes" ]]; then
+            /usr/local/bin/ctrmgr_tools.py tag-all
+        fi
+    fi
+}
+
+function stop_sonic_services()
+{
+    if [[ x"$ASIC_TYPE" != x"mellanox" ]]; then
+        debug "Stopping syncd process..."
+        docker exec -i syncd /usr/bin/syncd_request_shutdown --cold > /dev/null
+        sleep 3
+    fi
+}
+
+function clear_warm_boot()
```

Comment:
Function also did kernel unload, please update function name to reflect that. You can name it as clear_lingering_reboot_config